src/System Application/App/AppSource Gallery/src/AppSourceProductDetails.Page.al

```diff
@@ -0,0 +1,395 @@
+                {
+                    Caption = 'Popularity';
+                    ToolTip = 'Specifies the Popularity';
+                }
+                field(Rating_RatingAverage; AppSourceProductManager.GetStringValue(ProductObject, 'ratingAverage'))
+                {
+                    Caption = 'Rating Average';
+                    ToolTip = 'Specifies the Rating Average';
+                }
+                field(Rating_RatingCount; AppSourceProductManager.GetStringValue(ProductObject, 'ratingCount'))
+                {
+                    Caption = 'Rating Count';
+                    ToolTip = 'Specifies the Rating Count';
+                }
+            }
+
+
+            group(LinksGroup)
+            {
+                Caption = 'Links';
+
+                field(Links_LegalTermsUri; AppSourceProductManager.GetStringValue(ProductObject, 'legalTermsUri'))
+                {
+                    Caption = 'Legal Terms Uri';
+                    ToolTip = 'Specifies the Legal Terms Uri';
+                    ExtendedDatatype = Url;
+                }
+                field(Links_PrivacyPolicyUri; AppSourceProductManager.GetStringValue(ProductObject, 'privacyPolicyUri'))
+                {
+                    Caption = 'Privacy Policy Uri';
+                    ToolTip = 'Specifies the Privacy Policy Uri';
+                    ExtendedDatatype = Url;
+                }
+                field(Links_SupportUri; AppSourceProductManager.GetStringValue(ProductObject, 'supportUri'))
+                {
+                    Caption = 'Support Uri';
+                    ToolTip = 'Specifies the Support Uri';
+                    ExtendedDatatype = Url;
+                }
+            }
+        }
+    }
+
+    actions
+    {
+        area(Promoted)
+        {
+            actionref(Open_Promoted; OpenInAppSource) { }
+            actionref(Install_Promoted; Install) { }
+            actionref(Uninstall_Promoted; Uninstall) { }
+        }
+
+        area(Processing)
+        {
+            action(OpenInAppSource)
+            {
+                Caption = 'View in AppSource';
+                Scope = Page;
+                Image = Open;
+                ToolTip = 'View app in AppSource';
+
+                trigger OnAction()
+                begin
+                    AppSourceProductManager.OpenInAppSource(UniqueProductID);
+                end;
+            }
+
+            action(Install)
+            {
+                Caption = 'Install App';
+                Scope = Page;
+                Enabled = CurrentRecordCanBeInstalled;
+                Image = Insert;
+                ToolTip = 'Install App';
+
+                trigger OnAction()
+                var
+                    ExtensionManagement: Codeunit "Extension Management";
+                begin
+                    if not (CurrentRecordCanBeInstalled) then
+                        exit;
```

Comment:
Throw some error? It's not nice to fail silently.

Reply:
To address in this PR